compiler/rustc_mir_transform/src/check_unnecessary_transmutes.rs

```diff
@@ -0,0 +1,103 @@
+use rustc_middle::mir::visit::Visitor;
+use rustc_middle::mir::{Body, Location, Operand, Terminator, TerminatorKind};
+use rustc_middle::ty::{TyCtxt, UintTy};
+use rustc_session::lint::builtin::UNNECESSARY_TRANSMUTATION;
+use rustc_span::source_map::Spanned;
+use rustc_span::{Span, sym};
+use rustc_type_ir::TyKind::*;
```

Comment:
please don't rely on `rustc_type_ir` and use `ty::TyVariant`  instead. We intend to add a lint for that hopefully https://github.com/rust-lang/rust/issues/138449